flink-table/flink-table-common/src/main/java/org/apache/flink/table/sink/filesystem/FileSystemFileCommitter.java

```diff
@@ -0,0 +1,264 @@
+
+/**
+ * Default file system {@link FileCommitter} implementation. It just move all files to
+ * output path from temporary path.
+ *
+ * <p>NOTE: Here, we have no way to update catalog.
+ */
+@Internal
+public class FileSystemFileCommitter extends FileCommitter {
+
+	private static final Pattern PARTITION_NAME_PATTERN = Pattern.compile("([^/]+)=([^/]+)");
+
+	private final Path outputPath;
+	private final Map<String, String> staticPartitions;
+	private final String[] partitionColumns;
+
+	public FileSystemFileCommitter(
+			boolean overwrite,
+			Path tmpPath,
+			Path outputPath,
+			Map<String, String> staticPartitions,
+			String[] partitionColumns) {
+		super(tmpPath, overwrite);
+		this.outputPath = outputPath;
+		this.staticPartitions = staticPartitions;
+		this.partitionColumns = partitionColumns;
+	}
+
+	@Override
+	public void deletePath(Path taskTmpDir) throws Exception {
+		FileSystem.get(taskTmpDir.toUri()).delete(taskTmpDir, true);
+	}
+
+	@Override
+	public void commit(long checkpointId) throws Exception {
+		FileSystem fs = FileSystem.get(temporaryPath.toUri());
+		for (FileStatus taskStatus : fs.listStatus(temporaryPath)) {
+			String name = taskStatus.getPath().getName();
+			if (name.startsWith("cp-")) {
+				long cpId = Long.parseLong(name.substring(3, name.length()));
+
+				// commit paths that less than current checkpoint id.
+				if (cpId <= checkpointId) {
+					for (FileStatus status : fs.listStatus(taskStatus.getPath())) {
+						if (status.getPath().getName().startsWith("task-")) {
+							commitTaskOutputFiles(status.getPath());
+						}
+					}
+
+					// clean checkpoint path
+					deletePath(taskStatus.getPath());
+				}
+			}
+		}
+	}
+
+	/**
+	 * Commit a task temporary path.
+	 */
+	private void commitTaskOutputFiles(Path path) throws Exception {
+		try {
+			boolean isPartitioned = partitionColumns.length > 0;
+			boolean isDynamicPartition = isPartitioned && partitionColumns.length != staticPartitions.size();
+
+			if (isPartitioned) {
+				if (isDynamicPartition) {
+					List<Tuple2<LinkedHashMap<String, String>, Path>> partitionSpecs =
+							searchPartSpecAndPaths(path, partitionColumns.length);
+					for (Tuple2<LinkedHashMap<String, String>, Path> spec : partitionSpecs) {
+						loadPartition(spec.f1, spec.f0);
+					}
+				} else {
+					LinkedHashMap<String, String> partSpec = new LinkedHashMap<>();
+					for (String partCol : staticPartitions.keySet()) {
+						partSpec.put(partCol, staticPartitions.get(partCol));
+					}
+					loadPartition(path, partSpec);
+				}
+			} else {
+				moveFiles(path, outputPath);
+			}
+		} finally {
+			deletePath(path);
+		}
+	}
+
+	/**
+	 * Load a single partition.
+	 */
+	private void loadPartition(Path srcDir, LinkedHashMap<String, String> partSpec) throws Exception {
+		moveFiles(srcDir, new Path(outputPath, makePartitionName(partSpec)));
+	}
+
+	/**
+	 * Moves all files under srcDir into destDir. Delete destDir all files first when overwrite.
```

Comment:
Moves files from srcDir to destDir. Delete files in destDir first when overwrite.